pkg/util/kube.go

```diff
@@ -419,6 +419,38 @@ func GetPodName(n, labelSelector string) (pod string, err error) {
 	return
 }
 
+// Check PodReady (Example: 2/2 container running on a pod) returns true if the pod is ready. Params: namespace, pod label and timeout to check ready pod.
+func CheckPodReady(n, labelSelector string, timeout int) (ready bool, err error) {
+	ready = false
+	for i := 0; i < timeout; i++ {
+		pod, err := GetPodName(n, labelSelector)
+		status, _ := Shell(`oc get pod %s -n istio-system | grep istiod | awk '{print $2}'`, pod)
```

Comment:
Since you have the exact pod name, which you retrieved in the previous line, you don't need grepping by "istiod". You can also simplify these 2 lines to a single command:
```
oc get pods -n istio-system -l app=istiod -o jsonpath='{.items[0].status.phase}'
```
This will return "Running" once it's ready.

Reply:
Changes made

Reply:
This approach can not be done because we are getting from the Bug 0/1 container but pahse = Running:

`frherrer@frherrer-mac smmr % oc get pods -n istio-system -l app=istiod -o jsonpath='{.items[0].status.phase}'
Running%
frherrer@frherrer-mac smmr % oc get pods -n istio-system -l app=istiod
NAME                            READY   STATUS    RESTARTS   AGE
istiod-basic-5dcd956c99-fqbks   0/1     Running   0          38s
The issue is that we get the phase, and the phase is running, but the containers are 0/1 because the readiness probe fails:
frherrer@frherrer-mac smmr % kubectl get events  -n istio-system --field-selector type!=Normal,involvedObject.name=istiod-basic-5dcd956c99-fqbks |tail -1
3m31s       Warning   Unhealthy   pod/istiod-basic-5dcd956c99-fqbks   Readiness probe failed: Get "http://10.129.3.132:8080/ready": dial tcp 10.129.3.132:8080: connect: connection refused
frherrer@frherrer-mac smmr %`

Reply:
Improved using: 
`		output, err := Shell(`oc get pods -n istio-system -l %s -o jsonpath='{range .items[*]}{.status.containerStatuses[*].ready}'`, selector)
`